<launch>
    <param name="use_sim_time" value="true"/>
    <!-- Parameters setup based on Turtlebot Gmapping tutorial-->
    <node pkg="cgr_gridmap_slam_ros" type="cgr_gridmap_slam_node" name="cgr_slam" output="screen">
        <param name="tf_publish_period" value="0.05"/>
        <param name="map_frame" value="map"/>
        <param name="odom_frame" value="odom"/>
        <param name="base_frame" value="base_link"/>
        <param name="map_render_interval" value="5.0"/>
        <param name="force_render" value="false"/>
        <param name="slam/linear_update" value="0.5"/>
        <param name="slam/angular_update" value="0.4"/>
        <param name="slam/particles" value="80"/>
        <param name="slam/x_map_min" value="-50.0"/>
        <param name="slam/x_map_max" value="50.0"/>
        <param name="slam/y_map_min" value="-50.0"/>
        <param name="slam/y_map_max" value="50.0"/>
        <param name="slam/map_resol" value="0.05"/>
        <param name="slam/occ_threshold" value="0.25"/>
        <param name="slam/resample_threshold" value="0.5"/>
        <param name="motion_model/srr" value="0.01"/>
        <param name="motion_model/srt" value="0.02"/>
        <param name="motion_model/str" value="0.01"/>
        <param name="motion_model/stt" value="0.02"/>

        <param name="laser/max_range" value="8.0"/>
        <param name="laser/max_usable_range" value="6.0"/>
        <param name="scanmatch/sigma" value="0.05"/>
        <param name="scanmatch/kernel_size" value="1"/>
        <param name="scanmatch/lstep" value="0.05"/>
        <param name="scanmatch/astep" value="0.05"/>
        <param name="scanmatch/iterations" value="6"/>
        <param name="scanmatch/lsigma" value="0.075"/>
        <param name="scanmatch/lskip" value="0"/>
        <param name="scanmatch/minimum_score" value="50.0"/>

    </node>

</launch>